### ISSUE TITLE
division by zero error

### ISSUE THREAD

**Issue by MichaelDBA** [darold/pgbadger#444]
Using the latest version, pgbadger, pgbadger 10.0 and the PG 11 beta 3, the pgbadger commanded ended on a division by zero error.  Last lines in html file are:

```
<div id="checkpointdistance_graph" class="jqplot-graph linegraph"></div>
<script type="text/javascript">
/* <![CDATA[ */
var checkpointdistance_graph_18_d1 = [[1537363200000, 0],[1537363500000, 0],[1537363800000, 0],[1537364100000, 0],[1537364400000, 0],[1537364700000, 0],[1537365000000, 0],[1537365300000, 0],[1537365600000, 0],[1537365900000, 0],[1537366200000, 0],[1537366500000, 0],[1537366800000, 0],[1537367100000, 0],[1537367400000, 0],[1537367700000, 0],[1537368000000, 0],[1537368300000, 0],[1537368600000, 0],[1537368900000, 0],[1537369200000, 0],[1537369500000, 0],[1537369800000, 0],[1537370100000, 0],[1537370400000, 0],[1537370700000, 0],[1537371000000, 0],[1537371300000, 0],[1537371600000, 0],[1537371900000, 0],[1537372200000, 0],[1537372500000, 0],[1537372800000, 0],[1537373100000, 0],[1537373400000, 0],[1537373700000, 0],[1537374000000, 0],[1537374300000, 0],[1537374600000, 0],[1537374900000, 0],[1537375200000, 0],[1537375500000, 0],[1537375800000, 0],[1537376100000, 0],[1537376400000, 0],[1537376700000, 0],[1537377000000, 0],[1537377300000, 0],[1537377600000, 0],[1537377900000, 0],[1537378200000, 0],[1537378500000, 0],[1537378800000, 0],[1537379100000, 0],[1537379400000, 0],[1537379700000, 0],[1537380000000, 0],[1537380300000, 0],[1537380600000, 0],[1537380900000, 0],[1537381200000, 0],[1537381500000, 0],[1537381800000, 0],[1537382100000, 0],[1537382400000, 0],[1537382700000, 0],[1537383000000, 0],[1537383300000, 0],[1537383600000, 0],[1537383900000, 0],[1537384200000, 0],[1537384500000, 0],[1537384800000, 0],[1537385100000, 0],[1537385400000, 0],[1537385700000, 0]];
var checkpointdistance_graph_18_d2 = [[1537363200000, 0],[1537363500000, 0],[1537363800000, 0],[1537364100000, 0],[1537364400000, 0],[1537364700000, 0],[1537365000000, 0],[1537365300000, 0],[1537365600000, 0],[1537365900000, 0],[1537366200000, 0],[1537366500000, 0],[1537366800000, 0],[1537367100000, 0],[1537367400000, 0],[1537367700000, 0],[1537368000000, 0],[1537368300000, 0],[1537368600000, 0],[1537368900000, 0],[1537369200000, 0],[1537369500000, 0],[1537369800000, 0],[1537370100000, 0],[1537370400000, 0],[1537370700000, 0],[1537371000000, 0],[1537371300000, 0],[1537371600000, 0],[1537371900000, 0],[1537372200000, 0],[1537372500000, 0],[1537372800000, 0],[1537373100000, 0],[1537373400000, 0],[1537373700000, 0],[1537374000000, 0],[1537374300000, 0],[1537374600000, 0],[1537374900000, 0],[1537375200000, 0],[1537375500000, 0],[1537375800000, 0],[1537376100000, 0],[1537376400000, 0],[1537376700000, 0],[1537377000000, 0],[1537377300000, 0],[1537377600000, 0],[1537377900000, 0],[1537378200000, 0],[1537378500000, 0],[1537378800000, 0],[1537379100000, 0],[1537379400000, 0],[1537379700000, 0],[1537380000000, 0],[1537380300000, 0],[1537380600000, 0],[1537380900000, 0],[1537381200000, 0],[1537381500000, 0],[1537381800000, 0],[1537382100000, 0],[1537382400000, 0],[1537382700000, 0],[1537383000000, 0],[1537383300000, 0],[1537383600000, 0],[1537383900000, 0],[1537384200000, 0],[1537384500000, 0],[1537384800000, 0],[1537385100000, 0],[1537385400000, 0],[1537385700000, 0]];


var series_arr = [ { label: "distance", color: "#6e9dc9" },{ label: "estimate", color: "#f4ab3a" } ];

create_download_button(18, 'btn btn-default');
var checkpointdistance_graph = create_linegraph('checkpointdistance_graph', 'Checkpoint mean distance and estimate (5 minutes period)', 'Number of bytes', series_arr, [checkpointdistance_graph_18_d1,checkpointdistance_graph_18_d2]);
add_download_button_event(18, 'checkpointdistance_graph');
/* ]]> */
</script>

                </div>
        </div><!-- end of Checkpoints Files -->


```

**Comment by MichaelDBA** [comment]
log file size =  5,605,787,885 bytes.
log_line_prefix = '%t [%p]: [%l-1] db=%d,user=%u,app=%a,client=%h '
command:  pgbadger ./postgresql-2018-09-19.log -d zeus_app_db  -j 8 -o ./pgbadger20180919.html

**Comment by MichaelDBA** [comment]
Is the 10.1 in the pg repos yet (I didn't see it this morning)?  Otherwise how to download the 10.1 version?

**Comment by darold** [comment]
Can you post the pgbadger error it gives the code line number where the problem occurs.

**Comment by MichaelDBA** [comment]
Thanks, Darold.  Here is the output with the line number:
POSTMAN11:/var/lib/pgsql>pgbadger ./t.log -d zeus_app_db  -j 8 -o ./pgbadger20180919.html
[========================>] Parsed 5605386576 bytes of 5605386576 (100.00%), queries: 1773805, events: 165
LOG: Ok, generating html report...
Illegal division by zero at /bin/pgbadger line 7728.

**Comment by darold** [comment]
ok I understand, you have to override pgbagder installed with the package installer:
```
cp /usr/local/bin/pgbadger  /bin/pgbadger
```
or use the full path in our command line `/usr/local/bin/pgbadger` until the RH package is updated to 10.1.

**Comment by MichaelDBA** [comment]
I'm closing this issue since the problem is fixed with 10.1.
